From bba75977358132f53d3b840462f122fed83f0ced Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:31 -0400
Subject: [tools] perf/top: Make -g refer to callchains

Message-id: <1408953210-26343-24-git-send-email-jolsa@redhat.com>
Patchwork-id: 88735
O-Subject: [PATCH RHEL7.1 BZ1133083 023/322] perf top: Make -g refer to callchains
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit bf80669e4f689f181f23a54dfe2a0f264147ad67
Author: David Ahern <dsahern@gmail.com>
Date: Thu Nov 14 20:51:30 2013 -0700

description
===========
In most commands -g is used for callchains. Make perf-top follow suit.
Move group to just --group with no short cut making it similar to
perf-record.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Documentation/perf-top.txt b/tools/perf/Documentation/perf-top.txt
index 7de01dd..cdd8d49 100644
--- a/tools/perf/Documentation/perf-top.txt
+++ b/tools/perf/Documentation/perf-top.txt
@@ -50,7 +50,6 @@ Default is to monitor all CPUS.
 --count-filter=<count>::
  Only display functions with more events than this.
 
--g::
 --group::
         Put the counters into a counter group.
 
@@ -143,12 +142,12 @@ Default is to monitor all CPUS.
 --asm-raw::
  Show raw instruction encoding of assembly instructions.
 
--G::
+-g::
  Enables call-graph (stack chain/backtrace) recording.
 
 --call-graph::
  Setup and enable call-graph (stack chain/backtrace) recording,
- implies -G.
+ implies -g.
 
 --max-stack::
  Set the stack depth limit when parsing the callchain, anything
diff --git a/tools/perf/builtin-top.c b/tools/perf/builtin-top.c
index 71e6402..531522d 100644
--- a/tools/perf/builtin-top.c
+++ b/tools/perf/builtin-top.c
@@ -1084,7 +1084,7 @@ int cmd_top(int argc, const char **argv, const char *prefix __maybe_unused)
        "dump the symbol table used for profiling"),
  OPT_INTEGER('f', "count-filter", &top.count_filter,
       "only display functions with more events than this"),
- OPT_BOOLEAN('g', "group", &opts->group,
+ OPT_BOOLEAN(0, "group", &opts->group,
        "put the counters into a counter group"),
  OPT_BOOLEAN('i', "no-inherit", &opts->no_inherit,
       "child tasks do not inherit counters"),
@@ -1105,7 +1105,7 @@ int cmd_top(int argc, const char **argv, const char *prefix __maybe_unused)
      " abort, in_tx, transaction"),
  OPT_BOOLEAN('n', "show-nr-samples", &symbol_conf.show_nr_samples,
       "Show a column with the number of samples"),
- OPT_CALLBACK_NOOPT('G', NULL, &top.record_opts,
+ OPT_CALLBACK_NOOPT('g', NULL, &top.record_opts,
       NULL, "enables call-graph recording",
       &callchain_opt),
  OPT_CALLBACK(0, "call-graph", &top.record_opts,
-- 
1.7.1